utils/python/CIME/SystemTests/system_tests_common.py

```diff
@@ -146,10 +223,11 @@ def _component_compare_test(self, suffix1, suffix2):
                                %(cmd, self._case.get_value('RUNDIR'), self._case.get_value('CASE'),
                                  self._case.get_value('CASEBASEID'), suffix1, suffix2, suffix1, suffix2))
         logger.debug("run %s results %d %s %s"%(cmd,rc,out,err))
-        if rc == 0:
-            append_status(out.replace("compare","compare functionality", 1) + "\n",
-                          sfile="TestStatus")
-        else:
+        status = TEST_PASS_STATUS if rc == 0 else TEST_FAIL_STATUS
+        with self._test_status:
+            self._test_status.set_status("%s_%s_%s" % (COMPARE_PHASE, suffix1, suffix2), status)
+
+        if rc != 0:
             append_status("Component_compare_test.sh failed out: %s\n\nerr: %s\n"%(out,err),
                           sfile="TestStatus.log")
             return False
```

Comment:
That's a great point. I didn't notice this because the TestStatus gets set correctly. I will change it so that it only returns success if there are no FAILs in the TestStatus file.